src/app/components/FrostedGlassPromo/index.test.tsx

```diff
@@ -118,6 +124,32 @@ describe('Frosted Glass Promo', () => {
     ).toBeInTheDocument();
   });
 
+  it('should render the appropriate elements - Optimo Promo No Image', () => {
+    const { container, getByText } = render(
+      <Component {...optimoPromoFixtureNoImage} />,
+    );
+
+    expect(getByText('2 febrero 2023'));
+    expect(
+      getByText(
+        optimoPromoFixture.item.headlines.promoHeadline.blocks[0].model
+          .blocks[0].model.text,
+      ),
+    );
+    expect(
+      container.querySelector(
+        `a[href="${makeRelativeUrlPath(
+          optimoPromoFixture.item.locators.canonicalUrl,
+        )}"]`,
+      ),
+    ).toBeInTheDocument();
+    expect(
+      container.querySelector(
+        `img[src="https://ichef.bbci.co.uk/news/400/cpsprodpb/36D1/production/_127933041__63970643_bbc-news-world-service-logo-nc.png.webp"]`,
+      ),
+    ).toBeInTheDocument();
+  });
```

Comment:
Just regarding this point from the ticket:

> Regardless of any behaviour stated above we need to engineer the fix to ensure that where the data is not in an expected format we omit the promo from the section it is included within (e.g. features and analysis); the current situation where it causes the whole page not to render is needs to be avoided in future.

Could we add a test for the scenario where the component returned is completely invalid? So deliberately leave out key information like the headline, url etc that we can't render a promo without.

What we are after is that whatever information is missing we don't throw an error but instead return `null` so we can render with the promo missing.

Reply:
Added test.